pkg/dns/dns.go

```diff
@@ -268,6 +301,97 @@ func (kd *KubeDNS) updateService(oldObj, newObj interface{}) {
 	kd.newService(newObj)
 }
 
+func assertIsIngress(obj interface{}) (*ext.Ingress, bool) {
+	if ing, ok := obj.(*ext.Ingress); ok {
+		return ing, ok
+	} else {
+		glog.Errorf("Type assertion failed! Expected object of type \"Ingress\", got %T", obj)
+		return nil, ok
+	}
+}
+
+func (kd *KubeDNS) newIngress(obj interface{}) {
+	if ing, ok := assertIsIngress(obj); ok {
+		glog.V(4).Infof("Add/Update for ingress %v", ing.Name)
+		subCache := NewTreeCache()
+		subCachePath := append(kd.domainPath, ingressSubdomain, ing.Namespace)
+
+		ips := []string{}
+		ingFQDN := strings.Join([]string{ing.Name, ing.Namespace, ingressSubdomain, kd.domain}, ".")
+		reverseRecord, _ := getSkyMsg(ingFQDN, 0)
+		for _, lbi := range ing.Status.LoadBalancer.Ingress {
+			// We generate an A record if the Ingress loadbalancer IP is set and a
+			// CNAME record if the hostname is set. If both IP and hostname fields
+			// are set we favor IP over the hostname.
+			// TODO(madhusudancs): We would like to generate an empty record
+			// (NODATA response) if neither of these fields are set to indicate
+			// that we know that the Ingress resource exists but we do not know
+			// what it points to. It is not entirely clear how to make that work
+			// with SkyDNS because setting an empty host field in the service
+			// record seems to return all the A records as response. Need to
+			// figure this out.
+			if len(lbi.IP) > 0 {
+				kd.newIngressRecord(subCache, ing.Namespace, ing.Name, lbi.IP)
```

Comment:
`kd.newIngressRecord()` calls `getSkyMsg()` which generates a log message for this. And it will be evident in that log that this will be an A record.